scripts/commands/promote.lua

```diff
@@ -9,35 +9,43 @@ cmdprops =
     parameters = "si"
 };
 
-function onTrigger(player, target, level)
-    if (level == nil) then
-        level = target;
-        target = player:getName();
-    end
+function error(player, msg)
+    player:PrintToPlayer(msg);
+    player:PrintToPlayer("@promote <player> <level>");
+end;
 
+function onTrigger(player, target, level)
+    local maxLevel = player:getGMLevel();
+    local targ;
+    
+    -- validate target
     if (target == nil) then
-        target = player:getName();
+        error(player, "You must provide a player name.");
+        return;
+    else
+        targ = GetPlayerByName(target);
+        if (targ == nil) then
+            error(player, string.format( "Player named '%s' not found!", target ) );
+            return;
+        end
     end
 
-    -- Validate the target..
-    local targ = GetPlayerByName( target );
-    if (targ == nil) then
-        player:PrintToPlayer( string.format( "Invalid player '%s' given.", target ) );
+    -- catch players trying to change level of equal or higher tiered GMs.
+    if (targ:getGMLevel() >= player:getGMLevel()) then
+        printf( "%s attempting to adjust same or higher tier GM %s.", player:getName(), targ:getName() );
+        targ:PrintToPlayer(string.format( "%s attempted to adjust your GM rank.", player:getName() ));
+        error(player, "You can not use this command on same or higher tiered GMs.");
         return;
     end
 
-    -- Validate the level..
-    if (level < 0) then 
-        level = 0;
+    -- validate level
+    if (level == nil or level < 0 or level > maxLevel) then
+        error(player, string.format("Invalid level.  Must be 0 to %i.", maxLevel ));
```

Comment:
gms should not be able to create additional gms of identical tier, only lesser gms. original command was designed that way intentionally.